lib/Doctrine/DBAL/Driver/SQLSrv/SQLSrvStatement.php

```diff
@@ -294,6 +296,15 @@ private function prepare()
                     ];
                     break;
 
+                case ParameterType::STRING:
+                    $params[$column - 1] = [
+                        &$variable,
+                        SQLSRV_PARAM_IN,
+                        SQLSRV_PHPTYPE_STRING(SQLSRV_ENC_CHAR),
+                        SQLSRV_SQLTYPE_VARCHAR('max'),
```

Comment:
It could be either `SQLSRV_SQLTYPE_VARCHAR` or `SQLSRV_SQLTYPE_NVARCHAR` — that's the challenge. I don't know the exact consequences of providing an incorrect type, but in my inderstanding:

1. If the column is `NVARCHAR` but the parameter is bound as `VARCHAR`, then a unicode value stored in the variable may be corrupted (requires research).
2. If the column is `VARCHAR` but the parameter is bound as `NVARCHAR`, then the DB may not be able to use the index by this field because it will have to cast the stored value to the binding type.

Before this change is accepted, we need to see what the potential issues are and that we're not introducing additional issues.